.github/workflows/java-ec2-asg-e2e-test.yml

```
@@ -155,12 +155,33 @@ jobs:
             fi
           done
 
+      - name: Wait for ASG instance to be ready
+        run: |
+          # Wait for up to 15 minutes
+          retry_counter=0
+          max_retry=90
+          while [ $retry_counter -lt $max_retry ]; do
+            instance_id=$(aws autoscaling describe-auto-scaling-groups --auto-scaling-group-names ec2-single-asg-${{ env.TESTING_ID }} --region ${{ env.E2E_TEST_AWS_REGION }} --query "AutoScalingGroups[].Instances[0].InstanceId" --output text)
+            status_ok=$(aws ec2 describe-instance-status --instance-ids $instance_id | jq '(.InstanceStatuses[] | select(.InstanceStatus.Status == "ok" and .SystemStatus.Status == "ok")) != {}')
+            
+            if [ "$status_ok" == "true" ]; then
+              echo "The instance is ready."
+              echo "INSTANCE_ID=$instance_id" >> $GITHUB_ENV
```

Let's be clearer here for future engineers
```suggestion
              echo "MAIN_SERVICE_INSTANCE_ID=$instance_id" >> $GITHUB_ENV
```